example/src/main/java/co/infinum/example/MainActivity.kt

```diff
@@ -6,106 +6,163 @@ import android.content.pm.PackageManager
-            prepareItems()
-            settingsRecyclerView.apply {
-                visibility = View.VISIBLE
-                layoutManager = LinearLayoutManager(this@MainActivity)
-                adapter = settingsAdapter
-            }
-        }
+//        settingsView.setOnClickListener {
+//            prepareItems()
+//            settingsRecyclerView.apply {
+//                visibility = View.VISIBLE
+//                layoutManager = LinearLayoutManager(this@MainActivity2)
+//                adapter = settingsAdapter
+//            }
+//        }
 
         takePictureView.setOnClickListener { _ ->
-            goldenEye.takePicture(
+            takePicture(
                 onPictureTaken = { bitmap ->
                     if (bitmap.width <= 4096 && bitmap.height <= 4096) {
                         displayPicture(bitmap)
                     } else {
                         reducePictureSize(bitmap)
                     }
-                },
-                onError = { it.printStackTrace() }
+                }
             )
         }
 
         recordVideoView.setOnClickListener { _ ->
             if (isRecording) {
                 isRecording = false
                 recordVideoView.setImageResource(R.drawable.ic_record_video)
-                goldenEye.stopRecording()
+                videoCapture?.stopRecording()
             } else {
                 startRecording()
             }
         }
 
         switchCameraView.setOnClickListener { _ ->
-            val currentIndex = goldenEye.availableCameras.indexOfFirst { goldenEye.config?.id == it.id }
-            val nextIndex = (currentIndex + 1) % goldenEye.availableCameras.size
-            openCamera(goldenEye.availableCameras[nextIndex])
+            lensFacing =
+                if (lensFacing == CameraSelector.LENS_FACING_BACK) CameraSelector.LENS_FACING_FRONT else CameraSelector.LENS_FACING_BACK
+            CameraX.getCameraWithLensFacing(lensFacing)
+            recreateCamera()
         }
     }
 
+    private fun recreateCamera() {
+        CameraX.unbindAll()
+        startCamera()
+    }
+
+    private fun takePicture(onPictureTaken: (Bitmap) -> Unit) {
+        // Get a stable reference of the modifiable image capture use case
+        val imageCapture = imageCapture ?: return
+
+        // Create timestamped output file to hold the image
+        val photoFile = File(
+            outputDirectory,
+            SimpleDateFormat(
+                FILENAME_FORMAT, Locale.US
+            ).format(System.currentTimeMillis()) + ".jpg"
+        )
+
+        // Create output options object which contains file + metadata
+        val outputOptions = ImageCapture.OutputFileOptions.Builder(photoFile).build()
+
+        // Setup image capture listener which is triggered after photo has
+        // been taken
+        imageCapture.takePicture(
+            outputOptions, ContextCompat.getMainExecutor(this), object : ImageCapture.OnImageSavedCallback {
```

Comment:
This seems like a more customizable method to call so it's a bit difficult to compare to GoldenEye, is it possible to do it with a simpler method or are you missing some data in that case?

Reply:
Not sure what you mean, there is a simpler method which does not save the image, but leaves us to do what we need with it.

Reply:
...and this is what GoldenEye does right now, is it not? It gives you the image through the `onPictureTaken` callback instead of saving it somewhere.

Reply:
Yes actually, when we use take picture from camerax with just capture and not save, we get ImageProxy which we need to cast to bitmap image and with GoldenEye it's already a bitmap. But the better thing is that camerax has those both function, one with more options for saving and one with just capturing the image